docs/architecture/modernize-desktop/example-migration-core.md

```diff
@@ -0,0 +1,272 @@
+
+There's a tool called API Analyzer (<https://docs.microsoft.com/dotnet/standard/analyzers/api-analyzer>) that
+helps you identify APIs that aren't compatible with your code.
+
+#### Use \#if directives
+
+If you need different execution paths when targeting .NET Framework and .NET Core, you should use compilation constants and add some \#if directives to your code to keep the same code base for both targets.
+
+#### Technologies not available on .NET Core
+
+Some technologies aren't available on .NET Core, such as:
+
+* AppDomains
+* Remoting
+* Code Access Security
+* WCF Server
+* Windows Workflow
+
+Therefore, you'll need to look for a replacement for those if you're using them in your application. You can read more about it in [Microsoft documentation](https://docs.microsoft.com/dotnet/core/porting/net-framework-tech-unavailable).
+
+#### Regenerate autogenerated clients
+
+If your application uses autogenerated code, such as a WCF client, you may need to regenerate this code to target .NET Core. Sometimes, you can find some missing references since they may not be included as part of the default .NET Core assemblies set. Using a tool like <https://apisof.net/>, you can easily locate the assembly the missing reference lives in and add it from NuGet.
+
+#### Rolling back package versions
+
+As a general rule, we've previously stated that you better update every single package version to be compatible with .NET Core. However, you can find that targeting an updated and compatible version of an assembly just doesn't pay off. If the cost of change isn't acceptable, you can consider rolling back package versions keeping the ones you use on .NET Framework. Although they may not be targeting .NET Core, they should work well unless they call some unsupported APIs.
+
+### Run and test
+
+Once you have your application building with no errors, you can start the last step of the migration by testing every functionality.
+
+In this final step, you can find several different issues depending on the complexity of your application and the dependencies and APIs you're using.
+
+For example, if you use configuration files (*app.config*), you may find some errors at run time like Configuration Sections not present. Using the `Microsoft.Extensions.Configuration` NuGet package should fix that error.
+
+Another reason for errors is the use of the `BeginInvoke` and `EndInvoke` methods because they aren't supported on .NET Core. They aren't supported on .NET Core because they have a dependency on Remoting, which doesn't exist on .NET Core. To solve this issue, try to use the `await` keyword when available or the <xref:System.Threading.Tasks.Task.Run%2A?displayProperty=nameWithType> method.
```

Comment:
```suggestion
Another reason for errors is the use of the `BeginInvoke` and `EndInvoke` methods because they aren't supported on .NET Core. They aren't supported on .NET Core because they have a dependency on Remoting, which doesn't exist on .NET Core. To solve this issue, try to use the `await` keyword (when available) or the <xref:System.Threading.Tasks.Task.Run%2A?displayProperty=nameWithType> method.
```